src/grammarconstraints/utils.jl

```diff
@@ -0,0 +1,7 @@
+"""
+	Returns the new index of a grammar rule bsed on the provided `mapping`.
+	If `rule` is NOT a key in `mapping`, the original `rule` is returned.
+"""
+function _get_new_index(rule::Int, mapping::AbstractDict{<:Integer,<:Integer})
+    get(mapping, rule, rule)
+end # TODO: do we need this function?
```

Comment:
If it's only used once or twice, I would opt to write the `get(...)` directly instead. Maybe leave a comment to explain it since it looks odd on first glance to see `rule` there twice.

Reply:
Done